### ISSUE TITLE
Object's id and object's prototype conflicts

### ISSUE THREAD

**Issue by yoo2001818** [gaearon/normalizr#32]
If an object's ID conflicts with object's prototype names, Instead of creating new Object, Normalizr will try to merge object with the prototype object.

Let's say there's a schema:
```js
export const Tag = new Schema('tags', {
  idAttribute: 'name'
});
```

If I provide this object to normalizr, it will try to override 'constructor' function with provided attributes:
```js
{
  name: 'constructor',
  ...
}
```

Because of this, Firefox throws 'name is read only' error while calling normalizr:
```
TypeError: "name" is read-only
Stacktrace:
mergeIntoEntity@webpack:///./~/normalizr/lib/index.js?:59:7
visitEntity@webpack:///./~/normalizr/lib/index.js?:83:3
visit@webpack:///./~/normalizr/lib/index.js?:94:12
...
```
It doesn't throw an error on Chrome, but it doesn't work as expected.
![2015-10-22 22-26-05](https://cloud.githubusercontent.com/assets/4210268/10666328/e959e1ec-790b-11e5-93fd-3966d490fe03.png)
As you can see, 'Object', `constructor.name`, is taking the place instead of 'constructor'.

It happens with any other Object's prototype attributes too, such as `toString`, `hasOwnProperty`, `isPrototypeOf`, ...

To resolve this problem, I think Normalizr will have to create 'blank' object without a prototype,
or check whether if the 'original' object is a prototype object and override them if required.

**Comment by yoo2001818** [comment]
Both Firefox and Node v4.2.1 are able to handle JSON with 'constructor' keys in it.
![2015-10-22 22-31-41](https://cloud.githubusercontent.com/assets/4210268/10666467/bfba85ac-790c-11e5-99a9-456109620071.png)
I think just overriding them would be fine.

Also, this can check if an object inherits a property: 
```
Object.getPrototypeOf(obj)[key] === obj[key]
```

I'll make a pull request to solve this problem.

**Comment by yoo2001818** [comment]
I've created a pull request fixing this issue: #33

**Comment by gaearon** [comment]
Fixed via #33, out in 1.3.1.
Thanks!
